suite-native/device/src/hooks/useDeviceConnect.tsx

```diff
@@ -68,17 +104,37 @@ export const useDeviceConnect = () => {
         }
     }, [dispatch, hideAlert, isUnacquiredDevice, showAlert, translate]);
 
+    useEffect(() => {
+        if (!isFirmwareVersionSupported(firmwareVersion, deviceModel)) {
+            showAlert({
+                title: translate('moduleDevice.unsupportedFirmware.title'),
+                description: translate('moduleDevice.unsupportedFirmware.description'),
+                icon: 'warningCircle',
+                pictogramVariant: 'red',
+                primaryButtonTitle: translate('moduleDevice.unsupportedFirmware.button'),
+                primaryButtonVariant: 'tertiaryElevation1',
+                appendix: <InstallFirmwareAppendix />,
+                onPressPrimaryButton: handleDisconnect,
+            });
+            dispatch(stopDiscoveryThunk());
```

Comment:
I guess the correct place to do this would be checking correct firmware version in discovery start thunk so it doesn't run at all for outdated firmwares? We don't let users inside the app so there's no point even starting it

Reply:
b17e3a181

Reply:
b17e3a181